Assets/BossRoom/Scripts/Client/Game/Character/ClientCharacterVisualization.cs

```diff
@@ -287,6 +293,56 @@ private void SetAppearanceSwap()
             }
         }
 
+        /// <summary>
+        /// Returns the value we should set the Animator's "Speed" variable, which has the following range
+        /// (as configured in the AnimatorController's blend tree):
+        ///     0   = idle
+        ///     0.5 = walking
+        ///     1   = sprinting
+        ///     1.5 = fast-sprinting
+        ///     2   = comically fast-sprinting.
+        /// </summary>
+        /// <remarks>
+        /// Values in between these positions will be blended by the animator controller (e.g. a value of .75 is animated
+        /// as something between a walk and a run), but because we have footstep clips at specific speeds, we prefer to use
+        /// the specific animation-speeds that line up with the footsteps: 0.5 = walking footsteps, 1.0 = running footsteps.
+        /// Higher than 1 = no footsteps at all (at present).
+        /// 
+        /// This is why we don't try to determine the character's "normal" speed ourselves (by dividing their current speed in
+        /// m_NetState.NetworkMovementSpeed with the character class's base movement speed). Doing so could give us more precise
+        /// footsteps when the character's speed is buffed an arbitrary amount (e.g. a 10% speed buff would make the legs move
+        /// 10% faster)... but at the cost of the footstep sounds being misaligned.
+        ///
+        /// For a cartoony-styled game, it works out better to just rely on what the server says we're doing
+        /// (as represented by NetworkLifeState and MovementStatus values).
+        /// </remarks>
+        /// <returns>speed value, between 0 and 2</returns>
+        private float GetVisualMovementSpeed()
+        {
+            if (m_NetState.NetworkLifeState.Value != LifeState.Alive)
+            {
+                return 0;
+            }
+
+            switch (m_NetState.MovementStatus.Value)
```

Comment:
That data should be data driven and not programmer owned. We should have scriptable object for these. These enums shouldn't be necessary either with that same thinking

Reply:
We think GameData/Character/Shared might be a good location.

Reply:
Created new SO `VisualizationConfiguration`. Created one instance of this SO and attached it to each of the character visualizations: player, imp, and boss.

*Implementation Notes:*
- Note that we still need the enum to exist, because it's how the server conveys status conditions to the client.
- I moved as many other members as possible out of `ClientCharacterVisualization` and into `VisualizationConfiguration`.
- We can now pre-calculate hashes for the animation vars/triggers/tags in `VisualizationConfiguration.OnValidate()` -- this is an improvement over the old calculate-hashes-at-runtime code.
- Since ClientCharacterVisualization has seen many revisions during the project, it needed a little cleaning up: reordered variables: first serialized fields, then public vars (and pseudo-vars), then private vars.
- Code cleanup: refactored `ClientCharacterVisualization.IsAnimating` from a pseudo-var into a regular function to better convey that this is a non-trivial calculation.